---
title: "BnT_revisions"
author: "James Whipman"
date: "2025-03-21"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Taking consecutive tonsil segments and celltyping
# Introduction
We segment single cells from classic IMC data and high resolution data and determine
how easy it is to phenotype cells. Whether classic IMC has double positive cells from 
misegmentation. 

```{r read in packages}
library(RANN)  
library(SingleCellExperiment)
library(ggplot2)
library(ggpubr)
library(CATALYST)
library(imcRtools)
library(scuttle)
library(BiocParallel)
library(reshape2)
library(circlize)
library(mclust)
library(ComplexHeatmap)
library(EBImage)
library(Rphenoannoy)
```

```{r Read in the necessary data}
input_folder <- "/mnt/central_nas/projects/Pladioc/SRIMC_project/Zenodo/Figure_2/BnT/"
cells <- read.csv(file = paste0(input_folder, "classic_bnt_objects.csv"), stringsAsFactors = FALSE)
panel <- read.csv(file = paste0(input_folder, "bnt_panel.csv"), stringsAsFactors = FALSE)
spillover <- read.csv("/mnt/central_nas/projects/Pladioc/SRIMC_project/Zenodo/Figure_2/BnT/bnt_spillover.csv", row.names = 1)

```

```{r meta data: cell objects}
# Add rownames (combination of image number and object number)
rownames(cells) <- paste0(cells$Image, "_", cells$Object)
cell_meta <- data.frame(CellNumber = cells$Object,
                       Center_X = cells$centroid.0,
                       Center_Y = cells$centroid.1,
                       Area = cells$area,
                       MajorAxisLength = cells$axis_major_length,
                       MinorAxisLength = cells$axis_minor_length, 
                       image = factor(cells$Image),
                       row.names = paste0(cells$Image, "_", cells$Object))

# check whether the order is the same
all.equal(rownames(cell_meta), rownames(cells))
cell_meta_classic <- as.data.frame(cell_meta)
cell_meta_classic$resolution <- "Classical"
colnames(cells)[c(3:5,7:36)] <- c("MPO", "H3", "CD11c", "ECAD","CCR7","VIM", "HLADR","CD4","SMA","CD11b","CD20","CD56","Decorin","CD3","DC_LAMP","ATP5A","Periostin","H3K27me2","HSP10","CD8","CD45", "FOXP3","CD68","Ki-67","CD8_b","FOXP3_b","IRF4","CD4","CD31","CXCL12","DNA1","DNA2","CD15")
cells_classic <- cells[, -c(1:2,6,37:42)]
```

```{r Input matched TMA}
sce_classic <- SingleCellExperiment(assays = list(counts = t(cells_classic)))
rownames(sce_classic) <- colnames(cells_classic)
colnames(sce_classic) <- rownames(cells_classic)
colData(sce_classic) <- DataFrame(cell_meta_classic)
```

```{r spillover correction}
rowData(sce_classic)$channel_name <- paste0(panel$channel[-4], "Di")
sce_classic <- compCytof(sce_classic, spillover, 
                 transform = TRUE, cofactor = 1,
                 isotope_list = isotope_list, 
                 overwrite = TRUE)

assay(sce_classic, "exprs") <- asinh(assay(sce_classic, "counts"))
assay(sce_classic, "scaled") <- t(scale(t(assay(sce_classic, "exprs"))))

cell_norm_counts <- t(apply(assay(sce_classic, "counts"), 1, function(x)(x-min(x))/(quantile(x, 0.99)-min(x))))
cell_norm_counts <- t(apply(cell_norm_counts, 1, function(x) pmin(x, 1)))
assay(sce_classic, "normalized", withDimnames = FALSE) <- cell_norm_counts
```


```{r transform so it aligns with the consecutive section}
transformation_matrix <- as.matrix(read.table("/mnt/central_nas/projects/Pladioc/SRIMC_project/Zenodo/Figure_2/BnT/bnt_transformation.txt", sep = ","))
coords <- as.matrix(cbind(sce_classic$Center_X, sce_classic$Center_Y, 1))
transformed_coords <- coords %*% as.matrix(t(transformation_matrix))
transformed_coords <- as.data.frame(transformed_coords)
sce_classic$transformed_X <- transformed_coords$V1
sce_classic$transformed_Y <- transformed_coords$V2
```

```{r Read in the necessary data}
input_folder <- "/mnt/central_nas/projects/Pladioc/SRIMC_project/Zenodo/Figure_2/BnT/"
cells <- read.csv(file = paste0(input_folder, "hrimc_bnt_objects.csv"), stringsAsFactors = FALSE)
panel <- read.csv(file = paste0(input_folder, "bnt_panel.csv"), stringsAsFactors = FALSE)
```

```{r meta data: cell objects}
# Add rownames (combination of image number and object number)
rownames(cells) <- paste0(cells$Image, "_", cells$Object)
cell_meta <- data.frame(CellNumber = cells$Object,
                       Center_X = cells$centroid.0,
                       Center_Y = cells$centroid.1,
                       Area = cells$area,
                       MajorAxisLength = cells$axis_major_length,
                       MinorAxisLength = cells$axis_minor_length, 
                       image = factor(cells$Image),
                       row.names = paste0(cells$Image, "_", cells$Object))

# check whether the order is the same
all.equal(rownames(cell_meta), rownames(cells))
cell_meta$resolution <- "HR-IMC"
colnames(cells)[c(3:5,7:36)] <- c("MPO", "H3", "CD11c", "ECAD","CCR7","VIM", "HLADR","CD4","SMA","CD11b","CD20","CD56","Decorin","CD3","DC_LAMP","ATP5A","Periostin","H3K27me2","HSP10","CD8","CD45", "FOXP3","CD68","Ki-67","CD8_b","FOXP3_b","IRF4","CD4","CD31","CXCL12","DNA1","DNA2","CD15")
cells_hi <- cells[, -c(1:2,6,37:42)]
```

```{r Input matched TMA}
sce_hr <- SingleCellExperiment(assays = list(counts = t(cells_hi)))
rownames(sce_hr) <- colnames(cells_hi)
colnames(sce_hr) <- rownames(cells_hi)
colData(sce_hr) <- DataFrame(cell_meta)
```


```{r spillover correction}
rowData(sce_hr)$channel_name <- paste0(panel$channel[-4], "Di")
sce_hr <- compCytof(sce_hr, spillover, 
                 transform = TRUE, cofactor = 1,
                 isotope_list = isotope_list, 
                 overwrite = TRUE)

assay(sce_hr, "exprs") <- asinh(assay(sce_hr, "counts"))
assay(sce_hr, "scaled") <- t(scale(t(assay(sce_hr, "exprs"))))

cell_norm_counts <- t(apply(assay(sce_hr, "counts"), 1, function(x)(x-min(x))/(quantile(x, 0.99)-min(x))))
cell_norm_counts <- t(apply(cell_norm_counts, 1, function(x) pmin(x, 1)))
assay(sce_hr, "normalized", withDimnames = FALSE) <- cell_norm_counts
```


#Cell phenotyping
Now we have two aligned consecutive sections, we seperately cluster and determine the amound of BnT cells

```{r remove non-overlapping cells}
set.seed(18987)
set1 <- data.frame(x = sce_classic$transformed_X, y = sce_classic$transformed_Y)
rownames
set2 <- data.frame(x =sce_hr$Center_X, y = sce_hr$Center_Y)

nn1 <- nn2(set2, set1, k = 1)
dist1 <- nn1$nn.dists[, 1]
set1_filtered <- set1[dist1 <= 20, ]

# Find nearest neighbor from set1 for each point in set2
nn2 <- nn2(set1, set2, k = 1)
dist2 <- nn2$nn.dists[, 1]
set2_filtered <- set2[dist2 <= 20, ]

# Keep only points present in both filtered sets
filtered_set1 <- set1[dist1 <= 20, ]
filtered_set2 <- set2[dist2 <= 20, ]

sce_classic <- sce_classic[,as.numeric(rownames(filtered_set1))]
sce_hr <- sce_hr[,as.numeric(rownames(filtered_set2))]
```


```{r check images are transformed}
print(plotSpatial(sce_hr,
          assay_type = "counts",
          img_id = "image",
          coords = c("Center_X", "Center_Y"),
          node_size_fix = 1,
          node_color_by = "CD20"))
```

To understand whether there is a regional effect of BnT cells, we select different regions of the image.
Inspection shows the presence the germinal center in part of the image which could skew the number of BnT cells recorded. To test this, we split the image into four zones. 

```{r double positivity}
x_mid <- median(sce_classic$transformed_X)
y_mid <- median(sce_classic$transformed_Y)

quadrant <- ifelse(sce_classic$transformed_X <= x_mid & sce_classic$transformed_Y <= y_mid, "Q1",
            ifelse(sce_classic$transformed_X >  x_mid & sce_classic$transformed_Y <= y_mid, "Q2",
            ifelse(sce_classic$transformed_X <= x_mid & sce_classic$transformed_Y >  y_mid, "Q3", "Q4")))
sce_classic$quadrant <- factor(quadrant)

x_mid <- median(sce_hr$Center_X)
y_mid <- median(sce_hr$Center_Y)

quadrant <- ifelse(sce_hr$Center_X <= x_mid & sce_hr$Center_Y <= y_mid, "Q1",
            ifelse(sce_hr$Center_X >  x_mid & sce_hr$Center_Y <= y_mid, "Q2",
            ifelse(sce_hr$Center_X <= x_mid & sce_hr$Center_Y >  y_mid, "Q3", "Q4")))
sce_hr$quadrant <- factor(quadrant)
```


```{r classic BnT}
markers <- c("MPO","CD11c","CD20","CD56","CD3", "DC_LAMP", "CD8", "CD68", "CD4","CD15")


mat <- t(assay(sce_classic, "normalized")[markers,])
set.seed(230619)
out <- Rphenoannoy(mat, k = 7)
clusters <- factor(membership(out[[2]]))
sce_classic$pg_clusters <- clusters

#Check cluster values to assign
sce_mean <- aggregateAcrossCells(sce_classic, ids = sce_classic$pg_clusters, statistics = "mean", use.assay.type = "counts")
assay(sce_mean, "exprs") <- asinh(assay(sce_mean, "counts"))
assay(sce_mean, "scaled") <- t(scale(t(assay(sce_mean, "exprs"))))
my_palette <- colorRamp2(c(-2.5, 0, 2.5), c("#204a77", "white",  "#A90C38"))
Heatmap(assay(sce_mean, "scaled"),
        col = my_palette,
        border = TRUE,
        name = "Scaled\nExpression")

table(sce_classic$pg_clusters)


clusters <- c("CD4 T-cell", "CD8 T-cell", "Macrophage" , "Dendritic cell", "B-cell",
              "BnT", "BnT", "CD8 T-cell", "Other", "CD4 T-cell",
              "CD4 T-cell", "BnT", "Other", "CD4 T-cell", "Neutrophil",
              "CD4 T-cell", "CD4 T-cell", "Dendritic cell", "Neutrophil", "Dendritic cell", 
              "B-cell","B-cell", "NK-cell", "Macrophage", "Neutrophil",
              "B-cell", "Dendritic cell", "B-cell", "B-cell")

names(clusters) <- as.character(c(1:29))

sce_classic$celltype <- clusters[sce_classic$pg_clusters]

sce_mean <- aggregateAcrossCells(sce_classic, ids = sce_classic$celltype, statistics = "mean", use.assay.type = "counts")
assay(sce_mean, "exprs") <- asinh(assay(sce_mean, "counts"))
assay(sce_mean, "scaled") <- t(scale(t(assay(sce_mean, "exprs"))))
my_palette <- colorRamp2(c(-2.5, 0, 2.5), c("#204a77", "white",  "#A90C38"))
Heatmap(assay(sce_mean, "scaled")[markers,],
        col = my_palette,
        border = TRUE,
        name = "Scaled\nExpression")
```



```{r Cell phenotyping hr-imc}
mat <- t(assay(sce_hr, "normalized")[markers,])
set.seed(230619)
out <- Rphenoannoy(mat, k = 7)
clusters <- factor(membership(out[[2]]))
sce_hr$pg_clusters <- clusters

markers <- c("MPO","CD11c","CD20","CD56","CD3", "DC_LAMP", "CD8", "CD68", "CD4","CD15")

#Check cluster values to assign
sce_mean <- aggregateAcrossCells(sce_hr, ids = sce_hr$pg_clusters, statistics = "mean", use.assay.type = "counts")
assay(sce_mean, "exprs") <- asinh(assay(sce_mean, "counts"))
assay(sce_mean, "scaled") <- t(scale(t(assay(sce_mean, "exprs"))))
my_palette <- colorRamp2(c(-2.5, 0, 2.5), c("#204a77", "white",  "#A90C38"))
Heatmap(assay(sce_mean, "scaled"),
        col = my_palette,
        border = TRUE,
        name = "Scaled\nExpression")


clusters <- c("CD4 T-cell", "CD4 T-cell", "BnT" , "CD4 T-cell", "Neutrophil",
              "CD4 T-cell", "Dendritic cell", "Dendritic cell", "Other", "Dendritic cell",
              "B-cell", "B-cell", "CD4 T-cell", "B-cell", "Other",
              "Macrophage", "NK-cell", "CD4 T-cell", "B-cell", "CD8 T-cell", 
              "Dendritic cell","CD4 T-cell", "Dendritic cell", "CD8 T-cell", "B-cell",
              "Dendritic cell", "B-cell", "B-cell", "B-cell")

names(clusters) <- as.character(c(1:28))

sce_hr$celltype <- clusters[sce_hr$pg_clusters]

sce_mean <- aggregateAcrossCells(sce_hr, ids = sce_hr$celltype, statistics = "mean", use.assay.type = "counts")
assay(sce_mean, "exprs") <- asinh(assay(sce_mean, "counts"))
assay(sce_mean, "scaled") <- t(scale(t(assay(sce_mean, "exprs"))))
my_palette <- colorRamp2(c(-2, 0, 2), c("#204a77", "white",  "#A90C38"))
Heatmap(assay(sce_mean, "scaled")[markers,],
        col = my_palette,
        border = TRUE,
        name = "Scaled\nExpression")
```


```{r merging the data}
sce_hr$transformed_X <- sce_hr$Center_X
sce_hr$transformed_Y <- sce_hr$Center_Y

set.seed(1099)
sce_joined <- cbind(sce_hr[,sample(colnames(sce_hr), 3500, replace = FALSE)], sce_classic[,sample(colnames(sce_classic), 3500, replace = FALSE)])
```

```{r}
sce_joined$celltype[which(sce_joined$celltype %in% c("Other", "NK-cell", "Neutrophil", "Macrophage", "Dendritic cell"))] <- "Other"

sce_joined
prevalence <- as.matrix(prop.table(table(sce_joined$image, sce_joined$celltype), margin = 2))
prevalence_palette <- setNames(c(alpha("#D6DFEC",0.5), alpha("#e7c28e",0.6)), unique(rownames(prevalence)))
annotation <- HeatmapAnnotation(Proportion = anno_barplot(t(prevalence), 
                                                      gp = gpar(fill = prevalence_palette), 
                                                      bar_width = 1,bar_length = 6,height= unit(1.5, "cm")))

legend <- Legend(labels = c("HR-IMC", "Classic"), title = "Proportion", legend_gp = gpar(fill = prevalence_palette))
sce_mean <- aggregateAcrossCells(sce_joined, ids = sce_joined$celltype, statistics = "mean", use.assay.type = "scaled")
assay(sce_mean, "scaled") <- t(scale(t(assay(sce_mean, "scaled")))) #Scale for visualization

draw(Heatmap(assay(sce_mean, "scaled")[c("CD8", "CD20", "CD3", "CD4"),],
        col = my_palette,
        border = TRUE,
        name = "Scaled\nExpression",
        top_annotation = annotation), 
        annotation_legend_list = list(legend))


```

```{r Quantifying BnT cells}
sce_hr$narrow <- sce_hr$celltype
sce_hr$narrow[!(sce_hr$narrow %in% c("B-cell", "BnT", "CD4 T-cell", "CD8 T-cell"))] <- NA

sce_classic$narrow <- sce_classic$celltype
sce_classic$narrow[!(sce_classic$narrow %in% c("B-cell", "BnT", "CD4 T-cell", "CD8 T-cell"))] <- NA

data <- t(rbind(unclass(prop.table((table(sce_hr$narrow, sce_hr$quadrant)), margin = 2))[2,], unclass(prop.table((table(sce_classic$narrow, sce_classic$quadrant)), margin = 2))[2,]))
colnames(data) <- c("HR-IMC", "Classic IMC")
data <- reshape2::melt(data)

ggplot(data, aes(x = Var2, y = value, fill = Var2)) +
  geom_bar(stat = "summary", fun = mean, width = 0.6) +
  geom_errorbar(stat = "summary", fun.data = mean_se, width = 0.2) +
  geom_jitter(width = 0, size = 2, alpha = 0.6) +  
  geom_line(aes(group = Var1), color = "gray50", alpha = 0.2) + 
  stat_compare_means(method = "t.test", paired = TRUE, comparisons = list(c("Classic IMC", "HR-IMC")),
label = "value", size = 4) +
  scale_fill_manual(values = c("#e7c28e", "#D7D7D7")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(y = expression("Proportion of BnT cells"),x = "Imaging technique")
```

#Pixel-level markers
Read in the images and compare the B and T double positive pixels

```{r read in the data}
path_to_folder <- "/mnt/central_nas/projects/Pladioc/SRIMC_project/Zenodo/Figure_2/BnT/images/pixel_level/"
path_to_image <- "/mnt/central_nas/projects/Pladioc/SRIMC_project/Zenodo/Figure_2/BnT/images/pixel_level/img/"   
panel <- read.csv(file = paste0(path_to_folder, "panel.csv"), stringsAsFactors = FALSE)
```

## sce object generation and data transformation
This function reads in the image files and applies Otsu thresholding to remove background pixels. Count data is arcsinh-transformed, corrected for spillover using a compensation matrix and arranged into an sce object.

```{r read in the images, perform spillover correction and convert to sce}
generate_sce <- function(path_to_image, panel, i) {
      images <- list.files(path_to_image)
      image <- EBImage::readImage(paste0(path_to_image, images[i]), all = TRUE, type = "tiff")
      grayscale_image <- image
      imageData(grayscale_image) <- asinh(imageData(grayscale_image))
      imageData(grayscale_image) <- apply(imageData(grayscale_image), c(1, 2), sum)
      threshold_otsu <- otsu(grayscale_image, range = c(min(imageData(grayscale_image)), max(imageData(grayscale_image))))
      foreground_indices <- as.data.frame(which(imageData(grayscale_image) > threshold_otsu, arr.ind = TRUE))
      foreground_indices$coord <- paste0(foreground_indices[,1], "_",foreground_indices[,2])      
                                         
      channel_data <- lapply(1:dim(image)[3], function(x) as.vector(imageData(image)[,,x]))
      count_matrix <- do.call(cbind, channel_data)
      coords <- expand.grid(row = 1:dim(image)[1], column = 1:dim(image)[2])
      coords$combo <- paste0(coords$row, "_",coords$column)

      counts <- as.data.frame(channel_data)
      colnames(counts) <- panel$name <- gsub("^[^_]*_", "", panel$name)
      sce <- SingleCellExperiment(assays = list(counts = t(counts)))
      sce$Center_X <- coords[,1]
      sce$Center_Y <- coords[,2]
      sce <- sce[,which(coords$combo %in% foreground_indices$coord)]
      sce$image <- gsub("\\.tiff$", "", images[i])
      rowData(sce)$channel_name <- paste0(panel$channel, "Di")
      isotope_list$ArAr <- 80
      sce <- compCytof(sce, spillover, 
                 transform = TRUE, cofactor = 1,
                 isotope_list = isotope_list, 
                 overwrite = TRUE)
      return(sce)
      }
```


```{r generate sce objects}
all_sce <- lapply(1:length(list.files(path_to_image)), function(x) generate_sce(path_to_image, panel, x))
sce <- do.call(cbind, all_sce)
remove(all_sce)
```

```{r}
rownames(sce)[c(1:3,5:34)] <- c("MPO", "H3", "CD11c", "ECAD","CCR7","VIM", "HLADR","None","SMA","CD11b","CD20","CD56","Decorin","CD3","DC_LAMP","ATP5A","Periostin","H3K27me2","HSP10","CD8","CD45", "FOXP3","CD68","Ki-67","None","FOXP3_b","IRF4","CD4","CD31","CXCL12","DNA1","DNA2","CD15")
```

```{r Double positive}
HR_sce <- sce[,which(sce$image == "HR_IMC")]
x_mid <- median(HR_sce$Center_X)
y_mid <- median(HR_sce$Center_Y)


quadrant <- ifelse(HR_sce$Center_X <= x_mid & HR_sce$Center_Y <= y_mid, "Q1",
            ifelse(HR_sce$Center_X >  x_mid & HR_sce$Center_Y <= y_mid, "Q2",
            ifelse(HR_sce$Center_X <= x_mid & HR_sce$Center_Y >  y_mid, "Q3", "Q4")))

# Add a new column based on the quadrant
HR_sce$quadrant <- factor(quadrant)

assay(HR_sce, "exprs") <- asinh(assay(HR_sce, "counts"))
assay(HR_sce, "scaled") <- t(scale(t(assay(HR_sce, "exprs"))))
cell_norm_counts <- t(apply(assay(HR_sce, "counts"), 1, function(x)(x-min(x))/(quantile(x, 0.99)-min(x))))
cell_norm_counts <- t(apply(cell_norm_counts, 1, function(x) pmin(x, 1)))
assay(HR_sce, "normalized", withDimnames = FALSE) <- cell_norm_counts


mcl.model <- densityMclust(t(assay(HR_sce, "scaled"))[,c("CD3")], 2)
HR_sce$CD3 = mcl.model$classification
mcl.model <- densityMclust(t(assay(HR_sce, "scaled"))[,c("CD20")], 2)
HR_sce$CD20 = mcl.model$classification
HR_sce$DP <- NA


HR_sce$DP[which(HR_sce$CD20 == "2" | HR_sce$CD3 == "2")] <- "Single positive"
HR_sce$DP[which(HR_sce$CD20 == "2" & HR_sce$CD3 == "2")] <- "Double positive"
```

To understand whether there is a regional effect of double positivity, we select different regions of the image.
Inspection shows the presence the germinal center in part of the image which could skew the number of DP cells if they are more tightly packed. To test this, we split the image into four zones. 

```{r}
Classic_sce <- sce[,which(sce$image == "Classic_IMC")]

x_mid <- median(Classic_sce$Center_X)
y_mid <- median(Classic_sce$Center_Y)


quadrant <- ifelse(Classic_sce$Center_X <= x_mid & Classic_sce$Center_Y <= y_mid, "Q1",
            ifelse(Classic_sce$Center_X >  x_mid & Classic_sce$Center_Y <= y_mid, "Q2",
            ifelse(Classic_sce$Center_X <= x_mid & Classic_sce$Center_Y >  y_mid, "Q3", "Q4")))
Classic_sce$quadrant <- factor(quadrant)

assay(Classic_sce, "exprs") <- asinh(assay(Classic_sce, "counts"))
assay(Classic_sce, "scaled") <- t(scale(t(assay(Classic_sce, "exprs"))))
cell_norm_counts <- t(apply(assay(Classic_sce, "exprs"), 1, function(x)(x-min(x))/(quantile(x, 0.99)-min(x))))
cell_norm_counts <- t(apply(cell_norm_counts, 1, function(x) pmin(x, 1)))
assay(Classic_sce, "normalized", withDimnames = FALSE) <- cell_norm_counts


mcl.model <- densityMclust(t(assay(Classic_sce, "scaled"))[,c("CD3")], 2)
Classic_sce$CD3 = mcl.model$classification
mcl.model <- densityMclust(t(assay(Classic_sce, "scaled"))[,c("CD20")], 2)
Classic_sce$CD20 = mcl.model$classification

Classic_sce$DP <- NA
Classic_sce$DP[which(Classic_sce$CD20 == "2" | Classic_sce$CD3 == "2")] <- "Single positive"
Classic_sce$DP[which(Classic_sce$CD20 == "2" & Classic_sce$CD3 == "2")] <- "Double positive"

```


```{r}
data <- t(rbind(unclass(prop.table((table(HR_sce$DP, HR_sce$quadrant)), margin = 2))[1,], unclass(prop.table((table(Classic_sce$DP, Classic_sce$quadrant)), margin = 2))[1,]))
colnames(data) <- c("HR-IMC", "Classic IMC")
data <- melt(data)

ggplot(data, aes(x = Var2, y = value, fill = Var2)) +
  geom_bar(stat = "summary", fun = mean, width = 0.6) +
  geom_errorbar(stat = "summary", fun.data = mean_se, width = 0.2) +
  geom_jitter(width = 0.1, size = 2, alpha = 0.6) +  # optional raw points
  stat_compare_means(method = "t.test", paired = FALSE, comparisons = list(c("Classic IMC", "HR-IMC")),
label = "value", size = 3) + 
  scale_fill_manual(values = c("#e7c28e", "#D7D7D7")) +
  theme_minimal() +
  theme(legend.position = "none") +
  labs(y = expression("Proportion of CD20"^"+" ~ "CD3"^"+" ~ " pixels"),x = "Imaging technique")
```


